src/ConfigTemplateDefinitionCollection.php

```diff
@@ -0,0 +1,184 @@
+ *
+ * @package Drupal\testsite_builder
+ */
+class ConfigTemplateDefinitionCollection {
+
+  /**
+   * The Yaml serializer.
+   *
+   * @var \Drupal\Component\Serialization\SerializationInterface
+   */
+  protected $yamlSerializer;
+
+  /**
+   * The collection ID.
+   *
+   * @var string
+   */
+  protected $id;
+
+  /**
+   * The entity type for this collection.
+   *
+   * @var string
+   */
+  protected $entityType;
+
+  /**
+   * The template definitions.
+   *
+   * @var array
+   */
+  protected $templates;
+
+  /**
+   * The directory where collection file is located.
+   *
+   * @var string
+   */
+  protected $directory;
+
+  /**
+   * The config template definition collection constructor.
+   *
+   * @param \Drupal\Component\Serialization\SerializationInterface $yaml_serializer
+   *   The Yaml serializer.
+   */
+  protected function __construct(SerializationInterface $yaml_serializer) {
+    $this->yamlSerializer = $yaml_serializer;
+  }
+
+  /**
+   * Creates instance of config template definition collection.
+   *
+   * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
+   *   The symfony container.
+   *
+   * @return \Drupal\testsite_builder\ConfigTemplateDefinitionCollection
+   *   The instance of config template definition collection.
+   */
+  protected static function create(ContainerInterface $container): ConfigTemplateDefinitionCollection {
+    return new static(
+      $container->get('serialization.yaml')
+    );
+  }
+
+  /**
+   * Create new config template definition collection from templates.
+   *
+   * @param string $id
+   *   The collection ID.
+   * @param string $entity_type
+   *   The entity type for this collection.
+   * @param array $templates
+   *   The list of raw config template definitions.
+   * @param string $directory
+   *   The directory where collection is located.
+   *
+   * @return \Drupal\testsite_builder\ConfigTemplateDefinitionCollection
+   *   Returns instance of config template definition collection.
+   */
+  public static function createFromTemplates(string $id, string $entity_type, array $templates, $directory): ConfigTemplateDefinitionCollection {
+    $instance = static::create(\Drupal::getContainer());
+
+    $instance->id = $id;
+    $instance->entityType = $entity_type;
+    $instance->templates = $templates;
+    $instance->directory = $directory;
+
+    return $instance;
+  }
+
+  /**
+   * Create instance from file.
+   *
+   * @param string $file_name
+   *   The config template collection configuration file.
+   *
+   * @return \Drupal\testsite_builder\ConfigTemplateDefinitionCollection
+   *   Returns instance of config template definition collection.
+   *
+   * @throws \Exception
+   */
+  public static function createFromFile($file_name): ConfigTemplateDefinitionCollection {
+    $instance = static::create(\Drupal::getContainer());
+    $collection = $instance->yamlSerializer->decode(file_get_contents($file_name));
```

Comment:
Ah, the static decode call again. Here as well as in line 178

Reply:
fixed #37